embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/session/SessionPayloadSourceImpl.kt

```diff
@@ -21,21 +23,32 @@ internal class SessionPayloadSourceImpl(
     private val spanSink: SpanSink,
     private val currentSessionSpan: CurrentSessionSpan,
     private val spanRepository: SpanRepository,
-    private val logger: InternalEmbraceLogger
+    private val logger: InternalEmbraceLogger,
+    private val sessionPropertiesServiceProvider: Provider<SessionPropertiesService>
 ) : SessionPayloadSource {
 
     override fun getSessionPayload(endType: SessionSnapshotType): SessionPayload {
+        val sharedLibSymbolMapping = captureDataSafely(logger) { nativeThreadSamplerService?.getNativeSymbols() }
+        val internalErrors = captureDataSafely(logger) { internalErrorService.currentExceptionError?.toNewPayload() }
+        val snapshots = retrieveSpanSnapshotData()
+
+        // Ensure the span retrieving is last as that potentially ends the session span, which effectively ends the session
+        val spans = retrieveSpanData(endType)
```

Comment:
This fixes a race condition that could cause a span snapshot of the NEXT session to be included in the payload for this session. Stopping the session span last ensures that doesn't happen.